R/Utilities.R

```diff
@@ -1275,104 +1286,109 @@ TADA_AutoFilter <- function(.data) {
+
+  # remove columns that are not required for TADA workflow
   print("TADA_Autofilter: removing columns not required for TADA workflow if they contain only NAs.")
-  
-  #create list of required columns that must be retained even if all values are NA
-  req.cols <- c("TADA.CharacteristicName",
-                "TADA.ResultSampleFractionText",
-                "TADA.MethodSpeciationName",
-                "TADA.ResultMeasure.MeasureUnitCode",
-                "TADA.ActivityMediaName",
-                "TADA.DetectionQuantitationLimitMeasure.MeasureUnitCode",
-                "TADA.ResultMeasureValueDataTypes.Flag",
-                "TADA.LatitudeMeasure",
-                "TADA.LongitudeMeasure",
-                "OrganizationFormalName",
-                "ActivityTypeCode",
-                "ActivityMediaName",
-                "MonitoringLocationTypeName",
-                "ActivityStartDateTime",
-                "CharacteristicName",
-                "ResultSampleFractionText",
-                "MethodSpeciationName",
-                "ResultMeasureValue",
-                "ResultMeasure.MeasureUnitCode",
-                "ResultDetectionConditionText",
-                "DetectionQuantitationLimitTypeName",
-                "DetectionQuantitationLimitMeasure.MeasureValue",
-                "DetectionQuantitationLimitMeasure.MeasureUnitCode",
-                "ResultDepthHeightMeasure.MeasureValue",
-                "ResultDepthHeightMeasure.MeasureUnitCode",
-                "ActivityRelativeDepthName",
-                "ActivityDepthHeightMeasure.MeasureValue",
-                "ActivityDepthHeightMeasure.MeasureUnitCode",
-                "ActivityTopDepthHeightMeasure.MeasureValue",
-                "ActivityTopDepthHeightMeasure.MeasureUnitCode",
-                "ActivityBottomDepthHeightMeasure.MeasureValue",
-                "ActivityBottomDepthHeightMeasure.MeasureUnitCode",
-                "CountryCode",
-                "StateCode",
-                "CountyCode",
-                "LatitudeMeasure",
-                "LongitudeMeasure",
-                "QAPPApprovedIndicator",
-                "QAPPApprovalAgencyName",
-                "ProjectFileUrl",
-                "MeasureQualifierCode",
-                "SampleCollectionEquipmentName", # required for continuous flag
-                "StatisticalBaseCode", # required for continuous flag
-                "ResultTimeBasisText", # required for continuous flag
-                "ResultValueTypeName", # required for continuous flag
-                "ActivityIdentifier",
-                "ProjectIdentifier",
-                "MonitoringLocationIdentifier",
-                "ResultIdentifier",
-                "OrganizationIdentifier")
-  
-  # create list of columns containing all NA values. 
-  na.cols <- .data %>% purrr::keep(~all(is.na(.x))) %>%
-   names()
-  
+
+  # create list of required columns that must be retained even if all values are NA
+  req.cols <- c(
```

Comment:
For example, this could be a place to just read in the list of required columns from the csv.